packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -150,6 +150,8 @@ export function createInsightsMiddleware<
           immediate: true,
         });
 
+        const viewEventCache: Record<string, boolean> = {};
+
```

Comment:
maybe it could be cleared on the next promise / timer?

Reply:
I think you can use the queryID as a key, it's probably more efficient than stringifying the whole payload. Wdyt?

Reply:
ah yes queryID would be better indeed!

Reply:
Unfortunately view isn't sent with queryID